app/src/organisms/ProtocolSetupParameters/ChooseNumber.tsx

```diff
@@ -0,0 +1,166 @@
+}
+
+export function ChooseNumber({
+  handleGoBack,
+  parameter,
+  setParameter,
+}: ChooseNumberProps): JSX.Element | null {
+  const { makeSnackbar } = useToaster()
+
+  const { i18n, t } = useTranslation(['protocol_setup', 'shared'])
+  const keyboardRef = React.useRef(null)
+  const [paramValue, setParamValue] = React.useState<string>(
+    String(parameter.value)
+  )
+
+  // We need to arbitrarily set the value of the keyboard to a string the
+  // same length as the initial parameter value (as string) when the component mounts
+  // so that the delete button operates properly on the exisiting input field value.
+  const [prevKeyboardValue, setPrevKeyboardValue] = React.useState<string>('')
+  React.useEffect(() => {
+    const arbitraryInput = new Array(paramValue).join('*')
+    // @ts-expect-error keyboard should expose for `setInput` method
+    keyboardRef.current?.setInput(arbitraryInput)
+    setPrevKeyboardValue(arbitraryInput)
+  }, [])
+
+  if (parameter.type !== 'int' && parameter.type !== 'float') {
+    console.log(`Incorrect parameter type: ${parameter.type}`)
+    return null
+  }
+  const handleClickGoBack = (newValue: number): void => {
+    if (error != null) {
+      makeSnackbar(t('value_out_of_range_generic'))
+    } else {
+      setParameter(newValue, parameter.variableName)
+      handleGoBack()
+    }
+  }
+
+  const handleKeyboardInput = (e: string): void => {
+    if (prevKeyboardValue.length < e.length) {
+      const lastDigit = e.slice(-1)
+      if (
+        !'.-'.includes(lastDigit) ||
+        (lastDigit === '.' && !paramValue.includes('.')) ||
+        (lastDigit === '-' && paramValue.length === 0)
+      ) {
+        setParamValue(paramValue + lastDigit)
+      }
+    } else {
+      setParamValue(paramValue.slice(0, paramValue.length - 1))
+    }
+    setPrevKeyboardValue(e)
+  }
+
+  const paramValueAsNumber = Number(paramValue)
+  const resetValueDisabled = parameter.default === paramValueAsNumber
+  const { min, max } = parameter
+  const error =
+    paramValue === '' ||
+    Number.isNaN(paramValueAsNumber) ||
+    paramValueAsNumber < min ||
+    paramValueAsNumber > max
+      ? t(`value_out_of_range`, {
+          min: parameter.type === 'int' ? min : min.toFixed(1),
+          max: parameter.type === 'int' ? max : max.toFixed(1),
+        })
+      : null
+
+  return (
+    <>
+      <ChildNavigation
+        header={i18n.format(parameter.displayName, 'sentenceCase')}
+        onClickBack={() => {
+          handleClickGoBack(paramValueAsNumber)
+        }}
+        buttonType="tertiaryLowLight"
+        buttonText={t('restore_default')}
+        onClickButton={() =>
+          resetValueDisabled
+            ? makeSnackbar(t('no_custom_values'))
+            : setParamValue(String(parameter.default))
+        }
+      />
+      <Flex
+        alignSelf={ALIGN_CENTER}
+        gridGap={SPACING.spacing48}
+        paddingX={SPACING.spacing40}
+        paddingBottom={SPACING.spacing40}
```

Comment:
```suggestion
        padding=`${SPACING.spacing16} {SPACING.spacing40} {SPACING.spacing40}`
```

![Screenshot 2024-04-10 at 3 52 24 PM](https://github.com/Opentrons/opentrons/assets/474225/dcef431e-3354-4ea7-9e8a-ddcb2f8077ed)